### DISCUSSION TITLE
Execute some code on main thread, before worker threads are created?

### DISCUSSION THREAD

**Discussion by donmccurdy** [avajs/ava#3158] [Q&A] [answered]
The codebase I'm testing relies on sharp.js, which has a [known limitation](https://sharp.pixelplumbing.com/install#worker-threads) with worker threads —

> On some platforms, including glibc-based Linux, the main thread must call `require('sharp')` before worker threads are created. This is to ensure shared libraries remain loaded in memory until after all threads are complete.

I was hoping I might get around this with Ava's `require` config option, but that seems to run in the worker thread, not the main thread, so it's too late. Does Ava provide some way to ensure a given bit of code runs on the main thread before workers are created, or will I need to disable worker threads for glibc-based Linux platforms?

***

*EDIT: Possibly related to https://github.com/avajs/ava/issues/3027.*

**Comment by novemberborn** [comment, marked as answer]
Unofficially, `ava.config.js` is imported in the main thread, so you can require it there.
